repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala

```diff
@@ -97,34 +107,129 @@ class UriStripperFilter @Inject()(configurationService: ConfigurationService)
             case Failure(exception) =>
               logger.warn("Unable to parse Location header. Location header is malformed URI", exception)
             case Success(uri) =>
-              val locationTokens = splitUriIntoTokens(uri.getPath)
+              val locationTokens = splitPathIntoTokens(uri.getPath)
 
-              // figure out where to add the token if we can
-              ((previousToken.filter(locationTokens.contains), nextToken.filter(locationTokens.contains)) match {
-                case (Some(foundToken), _) => Some(locationTokens.indexOf(foundToken) + 1)
-                case (_, Some(foundToken)) => Some(locationTokens.indexOf(foundToken))
-                case _ => None
-              }).foreach(locationTokens.insert(_, token.get))
+              findReplacementTokenIndex(locationTokens, previousToken, nextToken)
+                .foreach(locationTokens.insert(_, token.get))
 
-              val preUri = Option(uri.getScheme).map(_ + "://" + uri.getHost + (if (uri.getPort != -1) ":" + uri.getPort else ""))
-              val postUri = Option(uri.getQuery).filterNot(_.trim.isEmpty).map(QueryParamIndicator +)
               wrappedResponse.replaceHeader(
                 CommonHttpHeader.LOCATION.toString,
-                preUri.getOrElse("") + joinTokensIntoUri(locationTokens) + postUri.getOrElse(""))
+                replacePath(uri, joinTokensIntoPath(locationTokens)))
           }
         }
       }
 
+      val applicableLinkPaths = config.getLinkResource
+        .filter(resource => resource.getUriPathRegex.r.pattern.matcher(wrappedRequest.getRequestURI).matches)
```

Comment:
It is a little strange that there isn't a better way of doing this. Either a `matches` method on Scala's `Regex` class, or an operator.

I'd be down for adding Groovy's `~=` operator to our commons library. If you write it as an `implicit` extension of `String` you could use it like so:
`"/my/pattern" ~= wrappedRequest.getRequestURI`